src/SFA.DAS.ApprenticeCommitments/Application/Commands/CreateApprenticeship/CreateApprenticeshipCommandHandler.cs

```diff
@@ -1,27 +1,51 @@
+using MediatR;
+using SFA.DAS.ApprenticeCommitments.Application.Services;
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using MediatR;
-using SFA.DAS.ApprenticeCommitments.Application.Services;
 
 namespace SFA.DAS.ApprenticeCommitments.Application.Commands.CreateApprenticeship
 {
     public class CreateApprenticeshipCommandHandler : IRequestHandler<CreateApprenticeshipCommand>
     {
         private readonly ApprenticeCommitmentsService _apprenticeCommitmentsService;
+        private readonly CommitmentsV2Service _commitmentsService;
         private readonly ApprenticeLoginService _apprenticeLoginService;
 
-        public CreateApprenticeshipCommandHandler(ApprenticeCommitmentsService apprenticeCommitmentsService, ApprenticeLoginService apprenticeLoginService)
+        public CreateApprenticeshipCommandHandler(
+            ApprenticeCommitmentsService apprenticeCommitmentsService,
+            ApprenticeLoginService apprenticeLoginService,
+            CommitmentsV2Service commitmentsV2Service)
         {
             _apprenticeCommitmentsService = apprenticeCommitmentsService;
             _apprenticeLoginService = apprenticeLoginService;
+            _commitmentsService = commitmentsV2Service;
         }
 
-        public async Task<Unit> Handle(CreateApprenticeshipCommand command, CancellationToken cancellationToken)
+        public async Task<Unit> Handle(
+            CreateApprenticeshipCommand command,
+            CancellationToken cancellationToken)
         {
             var id = Guid.NewGuid();
-            await _apprenticeCommitmentsService.CreateApprenticeship(id, command.ApprenticeshipId, command.Email);
-            await _apprenticeLoginService.SendInvitation(id, command.Email);
+
+            var apprentice = await _commitmentsService.GetApprenticeshipDetails(
+                command.EmployerAccountId,
+                command.ApprenticeshipId);
+
```

Comment:
Can we verify that we should get the apprenticeship name from comitments/approvals. I don't mind doing it (in fact I can see some good reasons to do it), but it's not the source of truth for that data element.  And from memory commitments does call FAT to get it (but I can't remember if this is then stored)

Reply:
@roysbailey - What do you think?  On the one hand, we are already calling the Commitments API and it returns the course name.  On the other hand it is not the source of truth.  This job will be running soon after the approval event, so one would imagine the Commitments API is up-to-date at that point, maybe?